.github/workflows/build-and-push-public-ecr.yaml

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+name: public-ecr-image-build-and-push
+
+on:        
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'the image version to push'
+        required: true
+      ecrRepo:
+        description: 'the name of the image to push'
+        required: true
+      githubRepo:
+        description: 'the name of the repository where code is hosted (org/repo)'
+        required: true
+      pathToDockerfile:
+        description: 'the path within the repository to the folder containing the dockerfile'
+        required: true
+        default: "."
+  
+env:
+  ECR_REPO: aws-observability/${{ github.event.inputs.ecrRepo }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Secret Names From ECR Repo Name
+        run: |
+          NEW_ECR_ID=`echo "${{ github.event.inputs.ecrRepo }}" | sed "s/-/_/g" | sed "s/\(.*\)/\U\1/"`
```

Comment:
can we add a check in case NEW_ECR_ID is a empty string, which would set the name to RELEASE_KEY_ID, which is the default credential

Reply:
Will do